### PR TITLE
Allows shared views across tenants.

### PR DESCRIPTION
**Pull request by jtkech**
Without using the theme engine, it allows e.g to have for a given module (enabled in 2 tenants) to have a controller mvc view that uses a given layout e.g `Layout = _Layout1` that is defined in 2 different modules acting as pseudo themes. Then, the layout that will be used by a given tenant will depend on its enabled pseudo themes.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/SharedViewLocationExpanderProvider.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            lock (_synLock)
+            {
+                if (_modulesWithSharedViews == null)
+                {
+                    var orderedModules = _extensionManager.GetExtensions()
+                        .Where(e => e.Manifest.Type.Equals("module", StringComparison.OrdinalIgnoreCase))
+                        .Reverse();
+
+                    var modulesWithPageSharedViews = new List<IExtensionInfo>();
+                    foreach (var module in orderedModules)
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/SharedViewLocationExpanderProvider.cs]: I think we can do both "Pages/Shared" 7 "Views/Shared" in the same loop

File: src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/SharedViewLocationExpanderProvider.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                        .Features.Any(sf => sf.Id == f.Id)).Select(f => f.Extension.Id).Distinct().ToArray();
+
+                    var enabledExtensions = _extensionManager.GetExtensions()
+                        .Where(e => enabledIds.Contains(e.Id)).ToArray();
+
+                    var pageSharedViewsPath = "/Pages/Shared/{0}" + RazorViewEngine.ViewExtension;
+
+                    modulePageSharedViewLocations = _modulesWithSharedViews
+                        .Where(m => enabledExtensions.Any(e => e.Id == m.Id))
+                        .Select(m => '/' + m.SubPath + pageSharedViewsPath);
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/SharedViewLocationExpanderProvider.cs]:
```csharp
.Select(m => $"/{m.SubPath}/Pages/Shared/{0}");
```

File: src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/SharedViewLocationExpanderProvider.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                    .Features.Any(sf => sf.Id == f.Id)).Select(f => f.Extension.Id).Distinct().ToArray();
+
+                var enabledExtensions = _extensionManager.GetExtensions()
+                    .Where(e => enabledIds.Contains(e.Id)).ToArray();
+
+                var sharedViewsPath = "/Views/Shared/{0}" + RazorViewEngine.ViewExtension;
+
+                moduleSharedViewLocations = _modulesWithSharedViews
+                    .Where(m => enabledExtensions.Any(e => e.Id == m.Id))
+                    .Select(m => '/' + m.SubPath + sharedViewsPath);
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/SharedViewLocationExpanderProvider.cs]: Same thing here

File: src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/SharedViewLocationExpanderProvider.cs (hunk 1 above)
Comment (jtkech) [reply src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/SharedViewLocationExpanderProvider.cs]: Yes, you're right, thanks. Just added the `Page` part too quickly

**Pull request merged**